test/ILLink.RoslynAnalyzer.Tests/TestCaseUtils.cs

```diff
@@ -209,5 +260,102 @@ internal static void GetDirectoryPaths (out string rootSourceDirectory, out stri
 			rootSourceDirectory = Path.GetFullPath (Path.Combine (artifactsBinDir, "..", "..", "test", "Mono.Linker.Tests.Cases"));
 			testAssemblyPath = Path.GetFullPath (Path.Combine (artifactsBinDir, "ILLink.RoslynAnalyzer.Tests", configDirectoryName, tfm));
 		}
+
+		internal static Task VerifyCodeFix<TAnalyzer, TCodeFix> (
+			string source,
+			string fixedSource,
+			DiagnosticResult[] baselineExpected,
+			DiagnosticResult[] fixedExpected,
+			int? numberOfIterations = null)
+			where TAnalyzer : DiagnosticAnalyzer, new()
+			where TCodeFix : CodeFixProvider, new()
+		{
+			string analyzerAttribute = GetAttributeDefinition<TAnalyzer> ();
+			string codeFixAttribute = GetAttributeDefinition<TCodeFix> ();
+			var attributeDefinitions = analyzerAttribute;
+			if (string.Equals (analyzerAttribute, codeFixAttribute) == false) {
+				attributeDefinitions += codeFixAttribute;
+			}
+
+			var test = new CSharpCodeFixVerifier<TAnalyzer, TCodeFix>.Test {
+				TestCode = source + attributeDefinitions,
+				FixedCode = fixedSource + attributeDefinitions,
+			};
+			test.ExpectedDiagnostics.AddRange (baselineExpected);
+			var analyzerMSBuildPropertyOption = GetAnalyzerMSBuildPropertyOption<TAnalyzer> ();
+			test.TestState.AnalyzerConfigFiles.Add (
+						("/.editorconfig", SourceText.From (@$"
+is_global = true
+build_property.{analyzerMSBuildPropertyOption} = true"
+)));
+			if (numberOfIterations != null) {
+				test.NumberOfIncrementalIterations = numberOfIterations;
+				test.NumberOfFixAllIterations = numberOfIterations;
+			}
+			test.FixedState.ExpectedDiagnostics.AddRange (fixedExpected);
+			return test.RunAsync ();
+		}
+
+		internal static Task VerifyDiagnostic<TAnalyzer> (
+			string source,
+			DiagnosticResult[] baselineExpected)
+			where TAnalyzer : DiagnosticAnalyzer, new() =>
+			CSharpAnalyzerVerifier<TAnalyzer>.VerifyAnalyzerAsync (
+				source,
+				TestCaseUtils.UseMSBuildProperties (GetAnalyzerMSBuildPropertyOption<TAnalyzer> ()),
+				baselineExpected);
+
+		internal static DiagnosticResult GetDiagnosticResult<TAnalyzer> ()
+			where TAnalyzer : DiagnosticAnalyzer, new()
+		{
+			return CSharpAnalyzerVerifier<TAnalyzer>.Diagnostic (GetDiagnosticId<TAnalyzer> ());
+		}
+
+		private static Dictionary<Type, string> diagnosticDict = new Dictionary<Type, string>
```

Comment:
The JIT is pretty smart and can often turn `typeof(Generic) == typeof(X)` into a JIT-time constant expression. Worst case, for two or three items it's still much faster to do if-else.